crates/bevy_scene/src/serde.rs

```diff
@@ -352,14 +352,14 @@ impl<'a, 'de> Visitor<'de> for ComponentVisitor<'a> {
     {
         let mut added = HashSet::new();
         let mut components = Vec::new();
-        while let Some(key) = map.next_key::<&str>()? {
-            if !added.insert(key) {
+        while let Some(key) = map.next_key::<String>()? {
```

Comment:
This potentially allocates a lot. Especially combined with the clone below. Is there a way to opprotuninistically avoid the cloning? i.e. with `Cow<'static, str>`?

Reply:
Just tested it and it seems to work! I'll see about changing it for the other instances of this pattern as well.